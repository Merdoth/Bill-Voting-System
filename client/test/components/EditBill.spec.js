import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import { EditBill } from
  '../../js/components/EditBill';


window.$ = require('jquery');

/* global jest */
const props = {
  match: {
    params: () => { }
  },
  user: {
    token: {}
  },
  currentBill: {},
  getAllBills: () => { },
  updateBill: jest.fn()
};
const setup = () => shallow(<EditBill {...props} />);

describe('EditBill Component snapshot', () => {
  it('it should render the right amount of elements', () => {
    const wrapper = setup();
    expect(wrapper).toMatchSnapshot();
  });
});

describe('onSubmit()', () => {
  it('should update bill to the state', () => {
    const event = {
      preventDefault: jest.fn(),
    };
    const wrapper = setup();
    wrapper.setState({
      image: {}
    });
    const form = wrapper.find('form');
    form.simulate('submit', event);
    expect(props.updateBill).toHaveBeenCalled();
  });
});
describe('onChange()', () => {
  it('should set title to state when input values changes', () => {
    const event = {
      preventDefault: jest.fn(),
      target: {
        name: 'title',
        value: 'The big men in Nigeria'
      }
    };
    const wrapper = setup();
    wrapper.instance().onChange(event);
    expect(wrapper.instance().state.title).toBe('The big men in Nigeria');
  });
  it('should set billProgress to state when input values changes', () => {
    const event = {
      preventDefault: jest.fn(),
      target: {
        name: 'billProgress',
        value: 'Senate Voted'
      }
    };
    const wrapper = setup();
    wrapper.instance().onChange(event);
    expect(wrapper.instance().state.billProgress).toBe('Senate Voted');
  });
});
