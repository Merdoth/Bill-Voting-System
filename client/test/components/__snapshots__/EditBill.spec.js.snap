// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditBill Component snapshot it should render the right amount of elements 1`] = `
<div
  className="dashboard-container"
>
  <Connect(SideBar) />
  <main>
    <div
      className="welcome-caption"
    >
      <div
        className="mobile-hambuger"
      >
        <a
          className="button-collapse hide-on-large-only "
          data-activates="slide-out"
          href="#"
        >
          <i
            className="material-icons"
          >
            menu
          </i>
        </a>
      </div>
      <span
        className="float-header-with-flex"
      />
      <span>
        <h5
          className="username-format"
        >
          <i
            className="material-icons username-icon"
          >
            account_circle
          </i>
          Welcome, 
        </h5>
      </span>
    </div>
    <div
      className="container"
    >
      <div
        className="edit-wrapper"
      >
        <div>
          <h2>
            Edit bill
          </h2>
        </div>
        <div
          className="account-summary-section"
        >
          <div>
            <form
              className="col s12 update"
              onSubmit={[Function]}
            >
              <div
                className="row"
              >
                <div
                  className="input-field col s12 m12"
                >
                  <input
                    className="form-control"
                    id="title"
                    name="title"
                    onChange={[Function]}
                    pattern=".{5,50}"
                    required={true}
                    title="5 characters minimum"
                    type="text"
                    value=""
                  />
                  <label
                    className="control-label"
                    htmlFor="title"
                  >
                    Title :
                  </label>
                </div>
              </div>
              <div
                className="row"
              >
                <div
                  className="input-field col s12 m12"
                >
                  <input
                    className="form-control"
                    id="description"
                    name="description"
                    onChange={[Function]}
                    pattern=".{8,1000}"
                    required={true}
                    title="8 characters minimum and 1000 maximum"
                    type="text"
                    value=""
                  />
                  <label
                    className="control-label"
                    htmlFor="description"
                  >
                    Description :
                  </label>
                </div>
              </div>
              <div
                className="row"
              >
                <label>
                  Bill Progress:
                </label>
                <select
                  className="browser-default"
                  id="billProgress"
                  name="billProgress"
                  onChange={[Function]}
                  required="true"
                >
                  <option
                    defaultValue={true}
                    value="Not enacted"
                  >
                    Not enacted
                  </option>
                  <option
                    value="Senate Voted"
                  >
                    Senate Voted
                  </option>
                  <option
                    value="House passed"
                  >
                    House Passed
                  </option>
                </select>
              </div>
              <div
                className="row form-cta"
              >
                <button
                  className="btn shadow-effect"
                  href="#"
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>
`;
